src/lib/apis/fetch.ts

```diff
@@ -20,9 +20,26 @@ export const constructUrlAndParams = (method, url): URLAndRequestBodyParams => {
   if (method === "PUT" || method === "POST" || method === "DELETE") {
     const [path, queryParams] = url.split("?")
     const parsedParams = parse(queryParams)
+    let body
 
     if (isExisty(parsedParams)) {
-      opts.body = parsedParams
+      // If the query params are formatted in a 0, 1, 2 way we know it's an array
+      // disguised as an object. Convert it back to an array.
+      // TODO: Extract this into own PR and write a test
```

Comment:
Will triple check, but `stringify` will always convert to an object; the various settings deal with arrays nested within the object.

Reply:
This sounds like a bug in qs then? What's the query string look like?

Reply:
- Given `["foo", "bar"]
- `qs.stringify`, which [expects an object](https://github.com/ljharb/qs#stringifying), will transform this into `0=foo&1=bar`
- which then parses into `{ 0: "foo", 1: "bar" }`

Reply:
Doesn't look like it, because this was clearly a bug! There's no way we could have gone from `0=foo&1=bar&2=baz` to a successful POST or PUT to an endpoint that is expecting an array.

Reply:
Why would we pass a bare array as a query string (inherently an object representation), ever?

Reply:
This is how we [get data back and forth](https://github.com/artsy/metaphysics/blob/main/src/lib/loaders/api/loader_interface.ts#L21-L23) between our api-constructed loader endpoints and our underlying `fetch` wrapper. Whether or not this is correct is another matter.

Reply:
I guess I don't see how that would ever result in us serializing a bare array onto the query string or why we would have any services that would expect that? What bug is this actually fixing?

Reply:
But, in that case though it'd never be a query string since it's a PUT, it can just be the body.